src/gui/history/qgshistoryproviderregistry.h

```diff
@@ -222,9 +182,34 @@ class GUI_EXPORT QgsHistoryProviderRegistry : public QObject
 
     /**
      * Clears the history for the specified \a backend.
+     *
+     * \see historyCleared()
      */
     bool clearHistory( Qgis::HistoryProviderBackend backend );
 
+  signals:
+
+    /**
+     * Emitted when an \a entry is added.
+     *
+     * \since QGIS 3.32
+     */
+    void entryAdded( long long id, const QgsHistoryEntry &entry, Qgis::HistoryProviderBackend backend );
```

Comment:
Seeking clarification on your implementation vision here: say I make a custom history provider in python for a plugin, what would the Qgis::HistoryProviderBackend value be here?  A plugin wouldn't be able to append a backend. Wouldn't it be better to tie this to an history provider ID instead?

Reply:
The backend is options like local, project, remote database etc. It defines where the history entry should be stored, whereas the provider is the "group" the action is associated with (eg processing algorithms, db queries, map exports, ....)

You can retrieve the provider id from the QgsHistoryEntry object.

Reply:
Should there be a plugins backend then?

Reply:
Plugins are free to use the existing backends. They just make up their own provider id and get all the history storage for free!

Reply:
Got it, I was misunderstanding backends as being categories themselves, not storage location / dataset. Thanks for the clarification.